File: backend/NordicWebHub.Api/Models/ProjectRequest.cs
using NordicWebHub.Api.Models.Enums;

namespace NordicWebHub.Api.Models;

public class ProjectRequest
{
    public int Id { get; set; }

    public int CompanyId { get; set; }

    public Company Company { get; set; } = null!;

    public int ServicePackageId { get; set; }

    public ServicePackage ServicePackage { get; set; } = null!;

    public string CustomerId { get; set; } = string.Empty;

    public ApplicationUser Customer { get; set; } = null!;

    public string Title { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public string BudgetRange { get; set; } = string.Empty;

    public ProjectRequestStatus Status { get; set; } = ProjectRequestStatus.Pending;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Project? Project { get; set; }
}

File: frontend/nordicwebhub-client/src/types/projectRequest.ts
export type ProjectRequestStatus = 'New' | 'Reviewed' | 'Approved' | 'Rejected'

export type ProjectRequest = {
  id: number
  companyId: number
  companyName: string
  servicePackageId: number
  servicePackageName: string
  customerId: string
  customerEmail: string
  title: string
  description: string
  budgetRange: string
  status: ProjectRequestStatus
  createdAt: string
}

export type CreateProjectRequestDto = {
  servicePackageId: number
  title: string
  description: string
  budgetRange: string
}

export type UpdateProjectRequestStatusDto = {
  status: ProjectRequestStatus
}
